internal/provider/provider.go

```diff
@@ -20,28 +20,39 @@ func New() *schema.Provider {
 		Schema: map[string]*schema.Schema{
 			"domain": {
 				Type:        schema.TypeString,
-				Required:    true,
```

Comment:
This was never really required, because if we specify a DefaultFunc the attribute becomes optional.